lib/typeahead/index.js

```diff
@@ -95,15 +100,23 @@ var Typeahead = React.createClass({
 
   getOptionsForValue: function (value, options) {
     if (this._shouldSkipSearch(value)) {
-      return [];
+      return {
+        visible: [],
+        truncated: false
+      };
     }
 
     var searchOptions = this._generateSearchFunction();
     var result = searchOptions(value, options);
+    var resultLength = result.length;
     if (this.props.maxVisible) {
       result = result.slice(0, this.props.maxVisible);
     }
-    return result;
+
+    return {
```

Comment:
Yeah, I'm not really a fan either because it's messing with the public API of react-typeahead.